### PR TITLE
Source File: add retry on SSHException('Error reading SSH protocol banner')

### PR DESCRIPTION
**Pull request by grubberr**
## What
Try to improve this oncall https://github.com/airbytehq/oncall/issues/1954

There is a customer whose sync periodically failed on error:
```
paramiko.ssh_exception.SSHException: Error reading SSH protocol banner[Errno 104] Connection reset by peer
```

From logs, it looks like a transient problem for him.
I have added `max_tries=5` for this case specifically.

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

**Review by lazebnyi — APPROVED** [review]
LGTM. Need add unit tests.

File: airbyte-integrations/connectors/source-file/source_file/client.py [hunk 1]
```diff
@@ -92,10 +94,21 @@ def close(self):
             self._file.close()
             self._file = None
 
+    def backoff_giveup(self, error):
```
[the remark this answers is not in the document]
Comment (grubberr) [reply airbyte-integrations/connectors/source-file/source_file/client.py:97]: OMG not easy task I will try now :)

[the remark this answers is not in the document]
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-file/source_file/client.py:97]: You can check facebook and hubspot unit tests as a hint.

[the remark this answers is not in the document]
Comment (grubberr) [reply airbyte-integrations/connectors/source-file/source_file/client.py:97]: Added

**Pull request merged**